core/src/main/scala/org/http4s/EntityEncoder.scala

```diff
@@ -147,11 +147,23 @@ trait EntityEncoderInstances extends EntityEncoderInstances0 {
   implicit val byteArrayEncoder: EntityEncoder[Array[Byte]] =
     chunkEncoder.contramap(Chunk.bytes)
 
-  // TODO fs2 port this is where we miss ByteVector
-  /*
   implicit val byteBufferEncoder: EntityEncoder[ByteBuffer] =
-    chunkEncoder.contramap(ByteVector.view)
-   */
+    chunkEncoder.contramap { bb =>
+      val duplicate = bb.duplicate()
+      // This is triggered if puts have been done, then we want to flip to read
+      if (duplicate.remaining != duplicate.capacity()) {
+        duplicate.flip()
+      }
+      if (duplicate.hasArray) {
+        Chunk.bytes(duplicate.array, 0, duplicate.remaining)
```

Comment:
Sorry, forgot to add a useful link to the docs: https://docs.oracle.com/javase/7/docs/api/java/nio/ByteBuffer.html#arrayOffset()